src/extensions/extension-manager.ts

```diff
@@ -39,7 +47,13 @@ export class ExtensionManager {
 
   async load() {
     logger.info("[EXTENSION-MANAGER] loading extensions from " + this.extensionPackagesRoot)
-    await fs.ensureDir(path.join(this.extensionPackagesRoot, "node_modules"))
+    if (this.inTreeFolderPath !== this.inTreeTargetPath) {
```

Comment:
when this condition is false? (ie `this.inTreeFolderPath === this.inTreeTargetPath`)

Reply:
I'm not actually sure (I think it depends on an environment). This is just to safeguard our in-tree extensions (if these match we would remove sources).

Reply:
Other option is to check if `this.inTreeFolderPath` is read-only.